operator/cnp_event.go

```diff
@@ -50,7 +51,11 @@ func init() {
 // enableCNPWatcher waits for the CiliumNetowrkPolicy CRD availability and then
 // garbage collects stale CiliumNetowrkPolicy status field entries.
 func enableCNPWatcher(apiextensionsK8sClient apiextensionsclientset.Interface) error {
-	log.Info("Starting to garbage collect stale CiliumNetworkPolicy status field entries...")
+	enableCNPStatusUpdates := kvstoreEnabled() && option.Config.K8sEventHandover && !option.Config.DisableCNPStatusUpdates
```

Comment:
This doesn't look to be enough, we need to register the flags in the operator as well.(`operator/option/config.go`)

Reply:
:face_with_head_bandage: `initEnv()` calls `option.Config.Populate()`, isn't that the shared options for both daemon and operator?

Reply:
Or is this more like the old approach we're trying to get rid of?

Reply:
To be clear, I haven't explicitly tested this version. @christarazi helped me to test the version from PR #13137 . So this is definitely a piece of feedback we should check on to make sure the args are respected.

Reply:
we still need to register them in `operator/flags.go`

Reply:
Ah right, the actual CLI arg declaration for the operator was missing. The viper argument fetch from the common `Populate()` function per above would presumably then not work correctly. Looks like @pchaigno has resolved this now, thanks!